File: app/Models/Alert.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Alert extends Model
{
    protected $fillable = [
        'greenhouse_id',
        'telemetry_reading_id',
        'alert_rule_id',
        'variable',
        'title',
        'description',
        'severity',
        'value',
        'status',
        'source',
        'fingerprint',
        'last_triggered_at',
        'resolved_at',
    ];

    protected function casts(): array
    {
        return [
            'value' => 'decimal:2',
            'last_triggered_at' => 'datetime',
            'resolved_at' => 'datetime',
        ];
    }

    public function greenhouse(): BelongsTo
    {
        return $this->belongsTo(Greenhouse::class);
    }

    public function reading(): BelongsTo
    {
        return $this->belongsTo(TelemetryReading::class, 'telemetry_reading_id');
    }

    public function rule(): BelongsTo
    {
        return $this->belongsTo(AlertRule::class, 'alert_rule_id');
    }
}

File: lumatek_mobile/lib/features/alerts/domain/alert.dart
import '../../../shared/models/json_utils.dart';

/// Alerta generada por AlertEvaluationService. Refleja App\Models\Alert.
class Alert {
  const Alert({
    required this.id,
    required this.greenhouseId,
    required this.title,
    this.description,
    this.variable,
    required this.severity,
    required this.status,
    this.value,
    this.source,
    this.lastTriggeredAt,
    this.resolvedAt,
  });

  final int id;
  final int greenhouseId;
  final String title;
  final String? description;

  /// Métrica que disparó la alerta: temperature, soil_humidity, etc.
  final String? variable;

  /// critical | warning | info
  final String severity;

  /// new | viewed | resolved
  final String status;

  final double? value;
  final String? source;
  final DateTime? lastTriggeredAt;
  final DateTime? resolvedAt;

  bool get isNew => status == 'new';
  bool get isResolved => status == 'resolved';
  bool get isCritical => severity == 'critical';

  factory Alert.fromJson(Map<String, dynamic> json) {
    return Alert(
      id: asInt(json['id']),
      greenhouseId: asInt(json['greenhouse_id']),
      title: asStringOrNull(json['title']) ?? 'Alerta',
      description: asStringOrNull(json['description']),
      variable: asStringOrNull(json['variable']),
      severity: asStringOrNull(json['severity']) ?? 'info',
      status: asStringOrNull(json['status']) ?? 'new',
      value: asDoubleOrNull(json['value']),
      source: asStringOrNull(json['source']),
      lastTriggeredAt: asDate(json['last_triggered_at']),
      resolvedAt: asDate(json['resolved_at']),
    );
  }
}
